openedx/core/djangoapps/course_roles/data.py

```diff
@@ -0,0 +1,160 @@
+        "view_all_published_content",
+        _("View All Published Content"),
+        _("Can view all Published content, including published-not-yet-released."),
+    )
+    ACCESS_INSTRUCTOR_DASHBOARD = PermissionData(
+        "access_instructor_dashboard",
+        _("Access Instructor Dashboard"),
+        _("Can view Instructor Dashboard with Basic Course Information."),
+    )
+    ACCESS_DATA_DOWNLOADS = PermissionData(
+        "access_data_downloads",
+        _("Access Data Downloads"),
+        _("Can view Data Downloads tab and generate all report types."),
+    )
+    MANAGE_GRADES = PermissionData(
+        "manage_grades",
+        _("Manage Grades"),
+        _("Can view and edit grades in Student Admin tab, Gradebook, and Open Responses."),
+    )
+    VIEW_GRADEBOOK = PermissionData(
+        "view_gradebook", _("View Gradebook"), _("Can view gradebook but cannot download or edit anything.")
+    )
+    MANAGE_ALL_USERS = PermissionData(
+        "manage_all_users",
+        _("Manage All Users"),
+        _("Can add, remove, and change role for all members of the course team."),
+    )
+    MANAGE_USERS_EXCEPT_ADMIN_AND_STAFF = PermissionData(
+        "manage_users_except_admin_and_staff",
+        _("Manage Users Except Admin and Staff"),
+        _("Can add, remove, change role for members of the course team, EXCEPT Admins and Staff."),
+    )
+    MANAGE_DISCUSSION_MODERATORS = PermissionData(
+        "manage_discussion_moderators", _("Manage Discussion Moderators"), _("Can add and remove all moderator roles.")
+    )
+    MANAGE_COHORTS = PermissionData(
+        "manage_cohorts", _("Manage Cohorts"), _("Can add and remove learners in cohorts; Can add new cohorts.")
+    )
+    MANAGE_STUDENTS = PermissionData(
+        "manage_students",
+        _("Manage Students"),
+        _(
+            "Can manage batch enrollments of audit learners and beta testers; Can manage cohorts, "
+            "extensions, student admin, Insights, bulk email, special exams, and open responses."
+        ),
+    )
+    MODERATE_DISCUSSION_FORUMS = PermissionData(
+        "moderate_discussion_forums",
+        _("Moderate Discussion Forums"),
+        _(
+            "Can view, edit, or delete all posts; Can pin, close, or reopen posts; "
+            "Can endorse responses, reports/flags."
+        ),
+    )
+    MODERATE_DISCUSSION_FORUMS_FOR_A_COHORT = PermissionData(
+        "moderate_discussion_forums_for_a_cohort",
+        _("Moderate Discussion Forums for a Cohort"),
+        _(
+            "Can view, edit, or delete all posts; Can pin, close, or reopen posts; Can endorse responses, "
+            "reports/flags. (Limited to user's own cohort)."
+        ),
+    )
+    MANAGE_CERTIFICATES = PermissionData(
+        "manage_certificates",
+        _("Manage Certificates"),
+        _("Can generate, regenerate, make exceptions, and invalidate certificates."),
+    )
+    MANAGE_LIBRARIES = PermissionData(
```

Comment:
Why is managing Libraries a Course Role/Permission?

Reply:
This permission is for the v1 content libraries. It will not need to exist if v2 content libraries is live, migrations are completed, and v1 code is removed before this code goes live, but a permission was included in case v1 content libraries is still live when a new role built off these permissions is created. This permission will be removed once v1 content libraries is fully sunset.

Reply:
I see. So v1 libraries will definitely still be around when this rolls out (for the community, even if edx.org decides to do the upgrade before then).

I'd like to understand a little more what it means when someone has this permission, because I'm not clear on what it would mean to have permission to `MANAGE_LIBRARIES` for a specific course, when the library exists outside of that course.

1. When we're deciding whether someone is allowed to create an entirely new library, do we check this permission at the org level?
2. When we're editing an individual v1 content library, does this use the v1 library key as the course key (because through hackiness, the v1 library keys are a subclass of CourseKey)?

Reply:
Looking through the existing code, we found only [one location](https://github.com/openedx/edx-platform/pull/33609/files#diff-b5a3a76341bb82aaa89f50d2308dc139109693a029b1f61dbe0c84504efabc42R163) (so far) where the code checks if a user has the `LibraryUser` role. 
This permission will be added as a permission check where the role is currently checked and used to create the `LibraryUser` role in `course_roles`, but at this time we haven't identified any additional locations where we would add a check for the new permission. 

1. In `course_roles` permissions checks are not dependent on the level on which the permission was granted, but just that the user has that permission in the context in question. This means if the user has the `LibraryUser` role at the org-wide level, they have that permission for all courses within the org. It also means if we check if the user has the permission, we are checking for course, org, and instance wide assignment of the permission.
2. We will likely have to make a determination on that when we update the code referenced above. `course_roles` has a fk dependency so we cannot assign a `course_role` on the library itself. It is also possible that the migration order of roles from `student_courseaccessrole` to `course_roles_role` can be altered so that the `LibraryUser` role stays on the `student_courseaccessrole` until after the point in time when the named release is using v2 content libraries and this permission (and role) are no longer relevant.

Reply:
FWIW, I'd suggest leaving this behind in `student_courseaccessrole`, but I'm okay either way in the short term.